File: solutions/golang/977/solution.go
func sortedSquares(nums []int) []int {
	ans := make([]int, len(nums))
	left := 0
	right, pos := len(nums)-1, len(nums)-1
	for left <= right {
		numLeft := int(math.Pow(float64(nums[left]), 2))
		numRight := int(math.Pow(float64(nums[right]), 2))
		if numLeft > numRight {
			ans[pos] = numLeft
			left++
		} else {
			ans[pos] = numRight
			right--
		}
		pos--
	}
	return ans
}

File: solutions/cpp/977/solution.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int left = 0;
        int right = nums.size() - 1;
        vector<int> ans(nums.size());
        int pos = nums.size() - 1;
        while (left <= right) {
            int leftSquared = pow(nums[left], 2);
            int rightSquared = pow(nums[right], 2);
            if (leftSquared > rightSquared) {
                ans[pos] = leftSquared;
                left++;
            } else {
                ans[pos] = rightSquared;
                right--;
            }
            pos--;
        }

        return ans;
    }
};